addons/website/models/ir_http.py

```diff
@@ -127,8 +127,15 @@ def _generate_routing_rules(self, modules, converters):
         logger.debug("_generate_routing_rules for website: %s", website_id)
         rewrites = self._get_rewrites(website_id)
         self._rewrite_len.__cache__.add_value(self, website_id, cache_value=len(rewrites))
-
-        for url, endpoint in super()._generate_routing_rules(modules, converters):
+        middleware_endpoint = None
+        rules = super()._generate_routing_rules(modules, converters)
+        all_rules = list(rules)
+        for url, endpoint in all_rules:
+            if url == '/rewrite/conditional_redirect':
+                middleware_endpoint = endpoint
+                break
+
+        for url, endpoint in all_rules:
```

Comment:
If the need of this loop is to assign the endpoint it could have been done [here](https://github.com/odoo/odoo/pull/214511/files#diff-f444d9b99919dbdbf8a57e1492478cb518cd2f78ec07dfedebff81140210671bR138)

```suggestion
        for url, endpoint in all_rules:
             if url == '/rewrite/conditional_redirect':
                middleware_endpoint = endpoint
```